src/entity/spatial/Polygon.cpp

```diff
@@ -44,6 +44,30 @@ void Polygon::setGeometry(geom::Polygon* geometry)
     geometry_ = geometry;
 }
 
+void Polygon::setCoordinates(const std::vector<geom::Coordinate>& coordinates)
+{
+    geom::CoordinateSequence* seq = geom::CoordinateArraySequenceFactory::instance()->create();
+    seq->setPoints(coordinates);
+
+    setCoordinates(*seq);
+
+    delete seq;
+}
+
+void Polygon::setCoordinates(const geom::CoordinateSequence& seq)
+{
+    geom::LinearRing* ring = factory_->createLinearRing(seq);
+
+    std::vector< geom::Geometry* > holes;
+
+    auto polygon = factory_->createPolygon(*ring, holes);   //copy ring to avoid ownership issues
```

Comment:
But you just created the ring in this method, just to destroy it afterwards! So why not transfer the ownership to the new polygon?

Reply:
Because I didn't trust `geos::geom`. ;-)
But I tried another call for this successfully and will commit it soon.